### ISSUE TITLE
Tried unsuccessfully to get this to compile on macOS rosetta

### ISSUE THREAD

**Issue by o0101** [1fishe2fishe/EXODUS#6]
Unfortunately, it would have been good! After removing some 'progbits' asm statements from the make instructions and running brew install sdl2 cmake, I was left with:

```
[ 94%] Building C object src/exodus/CMakeFiles/exodus.dir/posix/dbg.c.o
In file included from /Users/cris/EXODUS/src/exodus/posix/dbg.c:25:
/Library/Developer/CommandLineTools/SDKs/MacOSX14.4.sdk/usr/include/ucontext.h:51:2: error: The deprecated ucontext routines require _XOPEN_SOURCE to be defined
#error The deprecated ucontext routines require _XOPEN_SOURCE to be defined
 ^
[ 96%] Building C object src/exodus/CMakeFiles/exodus.dir/posix/seth.c.o
[ 98%] Building C object src/exodus/CMakeFiles/exodus.dir/posix/shims.c.o
/Users/cris/EXODUS/src/exodus/posix/dbg.c:74:13: error: use of undeclared identifier 'regs'; did you mean 'gets'?
  BackTrace(regs[5] /*RBP*/, regs[15] /*RIP*/);
            ^~~~
            gets
/Library/Developer/CommandLineTools/SDKs/MacOSX14.4.sdk/usr/include/stdio.h:164:7: note: 'gets' declared here
char    *gets(char *);
         ^
/Users/cris/EXODUS/src/exodus/posix/dbg.c:74:13: error: subscript of pointer to function type 'char *(char *)'
  BackTrace(regs[5] /*RBP*/, regs[15] /*RIP*/);
            ^~~~
/Users/cris/EXODUS/src/exodus/posix/dbg.c:74:30: error: use of undeclared identifier 'regs'; did you mean 'gets'?
  BackTrace(regs[5] /*RBP*/, regs[15] /*RIP*/);
                             ^~~~
                             gets
/Library/Developer/CommandLineTools/SDKs/MacOSX14.4.sdk/usr/include/stdio.h:164:7: note: 'gets' declared here
char    *gets(char *);
         ^
/Users/cris/EXODUS/src/exodus/posix/dbg.c:74:30: error: subscript of pointer to function type 'char *(char *)'
  BackTrace(regs[5] /*RBP*/, regs[15] /*RIP*/);
                             ^~~~
/Users/cris/EXODUS/src/exodus/posix/dbg.c:78:33: error: use of undeclared identifier 'regs'; did you mean 'gets'?
  FFI_CALL_TOS_2(sym->val, sig, regs);
                                ^~~~
                                gets
/Users/cris/EXODUS/build/src/exodus/asm/exodus/callconv.h:8:88: note: expanded from macro 'FFI_CALL_TOS_2'
#define FFI_CALL_TOS_2(a, b, c) FFI_CALL_TOS_2_((void *)(a), (uint64_t)(b), (uint64_t)(c))
                                                                                       ^
/Library/Developer/CommandLineTools/SDKs/MacOSX14.4.sdk/usr/include/stdio.h:164:7: note: 'gets' declared here
char    *gets(char *);
         ^
6 errors generated.
make[2]: *** [src/exodus/CMakeFiles/exodus.dir/posix/dbg.c.o] Error 1
make[2]: *** Waiting for unfinished jobs....
In file included from /Users/cris/EXODUS/src/exodus/posix/seth.c:39:
/Library/Developer/CommandLineTools/SDKs/MacOSX14.4.sdk/usr/include/ucontext.h:51:2: error: The deprecated ucontext routines require _XOPEN_SOURCE to be defined
#error The deprecated ucontext routines require _XOPEN_SOURCE to be defined
 ^
/Users/cris/EXODUS/src/exodus/posix/seth.c:148:33: error: too many arguments to function call, expected 1, have 2
  pthread_setname_np(c->thread, buf);
  ~~~~~~~~~~~~~~~~~~            ^~~
/Library/Developer/CommandLineTools/SDKs/MacOSX14.4.sdk/usr/include/pthread.h:535:5: note: 'pthread_setname_np' declared here
int     pthread_setname_np(const char*);
        ^
/Users/cris/EXODUS/src/exodus/posix/seth.c:161:5: error: call to undeclared function 'Awake'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
    Awake(&c->is_sleeping);
    ^
/Users/cris/EXODUS/src/exodus/posix/seth.c:176:3: error: call to undeclared function 'Sleep'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
  Sleep(&c->is_sleeping, UINT32_C(1),
  ^
/Users/cris/EXODUS/src/exodus/posix/seth.c:176:3: note: did you mean 'sleep'?
/Library/Developer/CommandLineTools/SDKs/MacOSX14.4.sdk/usr/include/unistd.h:481:3: note: 'sleep' declared here
         sleep(unsigned int) __DARWIN_ALIAS_C(sleep);
         ^
/Users/cris/EXODUS/src/exodus/posix/seth.c:208:45: error: use of undeclared identifier 'RegRbp'
  FFI_CALL_TOS_1_CUSTOM_BP(c->profiler_int, RegRbp, RegRip);
                                            ^
/Users/cris/EXODUS/src/exodus/posix/seth.c:208:53: error: use of undeclared identifier 'RegRip'
  FFI_CALL_TOS_1_CUSTOM_BP(c->profiler_int, RegRbp, RegRip);
                                                    ^
6 errors generated.
make[2]: *** [src/exodus/CMakeFiles/exodus.dir/posix/seth.c.o] Error 1
/Users/cris/EXODUS/src/exodus/posix/shims.c:280:7: warning: unused variable 'max' [-Wunused-variable]
  u64 max = UINT32_MAX >> 1;
      ^
/Users/cris/EXODUS/src/exodus/posix/shims.c:334:1: warning: non-void function does not return a value [-Wreturn-type]
}
^
2 warnings generated.
make[1]: *** [src/exodus/CMakeFiles/exodus.dir/all] Error 2
make: *** [all] Error 2
```

In case anyone knows any easy fix, or simply as ref for future travelers. I don't think there's an easy fix

**Comment by 1fishe2fishe** [comment]
Is this on Apple Silicon or x86 mac?

If it's an Apple Silicon mac, this project only supports x86_64. You can try [aiwinos](https://github.com/nrootconauto/aiwnios) but even that doesn't run on MacOS, it just supports ARM, so you'll still need to install Asahi Linux on your computer to run it.

If it's x86 mac, I have no plans on supporting MacOS as the main challenge in MacOS is that you can't mmap read+write+execute permissions on a page at the same time for "security" (also I don't own a mac), it's the same reason I don't support OpenBSD. I can have workarounds like memfd and mapping it twice but that's still impossible as I need them to be allocated on the lower 32 bits and would basically have to rewrite the TempleOS compiler and either slow down the whole thing (always mov into a register and call) or be really smart and dynamic with calling functions, and I'm not doing that.

**TLDR**: MacOS, on all architectures, is not supported and never will be. Just run TempleOS on a virtual machine, you guys have Parallels and all that jazz.

**Comment by o0101** [comment]
Hey @1fishe2fishe thanks for that! Sorry my comment should have been prefaced with a big "I know macOS is not officially supported"!! I apologize for that. I know it's not supported. I was just fingers crossing hoping, without having a clue as to all the internal and important reasons why this is not supported -- but I did assume there were valid and important reasons why it's not, thank you for elucidating those -- even tho it goes over my head a little bit. It's interesting! And I really appreciate your comment! Sorry for seeming to presume you should support macOS - i definitely wasn't! Appreciate your reply, thank you! :)

In answer to your question, I just, um...removed any "problematic" bits of CMake until I got it to run. Hahaha! :) 😊 😮

**Comment by 1fishe2fishe** [comment]
Sorry if my comment came off as a bit annoyed - I've dealt with way too many people with "dude make this run on my system NOW" and not really caring to know why it won't.
Thanks for trying out EXODUS though, maybe see you next time in another occasion.

**Comment by o0101** [comment]
I will def try to run it on a VM! I was just lazy/curious. Hahah, yes I know the type on GitHub. Sorry for coming across that way. The demanding requester: "It didn't work. It's broken. This sucks" or like you say "dude make this run on my system NOW" Haha. It's all good I thought that must've been it, and really I should have been clear up front I wasn't that - would have saved us all a little bit of grief. Good to see you here! It's weird that having GitHub repos is like customer service, both ways. Thank you :)
